### PR TITLE
[fix][monitor] Fix reporting pulsar_subscription_blocked_on_unacked_messages metric

### PR DESCRIPTION
**Pull request by aymkhalil**
### Motivation

The pulsar_subscription_blocked_on_unacked_messages metric was not scrapped by Prometheus although the dispatcher can be blocked on number of unacked messages:
```
Dispatcher read is blocked due to unackMessages 200010 reached to max 200000
```

### Modifications

When Prometheus scrapes stats topics, the previous logic was relying on the consumer flag to update the subscription  although the subscription configs and logic is independent of the attached consumer. The related config for subscription is `maxUnackedMessagesPerSubscription` and for consumers is `maxUnackedMessagesPerConsumer`

This change just copied over the value from the `SubscriptionStats` to the `AggregatedSubscriptionStats`

### Verifying this change

- [X] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

I couldn't add unit tests for `NamespaceStatsAggregator` in a short time but tested the changes locally with Pulsar standalone + Prometheous:

image

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: https://github.com/aymkhalil/pulsar/pull/3

### CODE REVIEW

**Review by eolivelli — APPROVED** [review]
good catch!

**Review by nicoloboschi — APPROVED** [review]

**Review by codelipenghui — CHANGES_REQUESTED** [review]
Nice catch!
@aymkhalil Could you please add a unit test to cover the new change?

**Comment by aymkhalil** [comment]
> 

Sure. PTAL. I bootstrapped a NamespaceStatsAggregatorTest class and added the related metrics in the PR.

**Review by codelipenghui — APPROVED** [review]

**Review by Technoboy- — APPROVED** [review]

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/stats/prometheus/NamespaceStatsAggregatorTest.java [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+        Subscription subscription = Mockito.mock(Subscription.class);
+        Consumer consumer = Mockito.mock(Consumer.class);
+        ConsumerStatsImpl consumerStats = new ConsumerStatsImpl();
+        when(consumer.getStats()).thenReturn(consumerStats);
+        when(subscription.getConsumers()).thenReturn(List.of(consumer));
+        subscriptionsMaps.put("my-subscription", subscription);
+        SubscriptionStatsImpl subStats = new SubscriptionStatsImpl();
+        TopicStatsImpl topicStats = new TopicStatsImpl();
+        topicStats.subscriptions.put("my-subscription", subStats);
+        when(topic.getStats(false, false, false)).thenReturn(topicStats);
```
Comment (asafm) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/stats/prometheus/NamespaceStatsAggregatorTest.java:102]: So whenever I add code to `NamespaceStatsAggregator` that calls to PersistentTopic (which will happen every time I have a new topic granularity metric), this test will fail. Mocking should be used cautiously, and when you mock way too much stuff, you should stop and think about what can be done differently. 

This test should have been deleted and added in `DispatcherBlockConsumerTest.testBlockDispatcherStats`

@aymkhalil @Technoboy- @codelipenghui

**Pull request merged**